### DISCUSSION TITLE
Suggestion: Make it easier to know whether project is a valid mixed python/rust project

### DISCUSSION THREAD

**Discussion by dhruv-ahuja** [PyO3/maturin#1843] [Ideas]
Hello guys, first of all, great job on the project! I recently used Maturin and Pyo3 for developing Rust bindings for my CLI song downloader app. It was generally a straightforward and pleasant experience. 

However, since I had started coding first without trying to build a wheel, I did not realize that my Rust code was not being included in the wheel. I think that there should be an easier way of knowing whether the project that we're building is considered to be a valid mixed python/rust project or not.

Currently we only have a message `🍹 Building a mixed python/rust project` during the build phase when running `maturin build/maturin develop`. 
Although the docs are very clear in this regard, I think adding a warning or info message, or tweaking the current message could further help and this could be colour-coded. Perhaps something like `You're building a Rust-only project, if this is a mixed Python/Rust project, please ensure that 'lib.name' matches Python directory name`.

It's possible that I'm the outlier here, since I do realize that the docs are really well-written and I did not have any issues whatsoever beyond this one. Please let me know your thoughts on the same.

Thank you!

**Comment by messense** [comment]
Thanks for the positive feedback.

I'm not convinced that changing the message would help avoiding this issue, maybe it's better if we make project layout an explict option in `pyproject.toml` so we can validate and fail the build when it mismatches the detected project layout.

**Reply by dhruv-ahuja** [reply]
True, that seems to be a much better way of handling this
